### PR TITLE
fix(MOPLAT-196): fix app restarting when taking photo

### PR DESCRIPTION
**Pull request by lordkiz**
The type of this PR is: **Bugfix**

This PR is about MOPLAT-196

### Description
The app restarts when the user takes a picture to pass to `react-native-image-crop-picker`. We do not know exactly why this is happening. And it seems to happen on random devices, but mostly on android-10 and android-11s. This hack silently clears the cache on android before taking the photo.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- fix app restarting when taking photo - kizito

</details>

[MOPLAT-196]: https://artsyproduct.atlassian.net/browse/MOPLAT-196?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (fix app restarting when taking photo - kizito)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0479b8fb7ed7c65a438ef372e757eebec2cd0268
</p>

File: HACKS.md [hunk 1]
```diff
@@ -275,3 +275,13 @@ If it is still an issue with old native navigation gone this can either be remov
 
 City Guide is a mixture of native components and react components, prop updates from the native side are not updating the component on the react native side without this manual check and update. See the PR here for the change in the AppRegistry:
 https://github.com/artsy/eigen/pull/6348
+
+# `React-Native-Image-Crop-Picker` App restarting when photo is taken. Fix is in `ArtsyNativeModule.clearCache`.
+
+#### When can we remove this:
+
+When we fix the actual issue.
```
Comment (pvinis) [inline HACKS.md:283]:
```suggestion
When we fix the actual issue. https://artsyproduct.atlassian.net/browse/MOPLAT-196
```
maybe we can link to this?

File: src/app/NativeModules/ArtsyNativeModule.tsx [hunk 2]
```diff
@@ -42,6 +42,12 @@ export const ArtsyNativeModule = {
           console.error("lockActivityScreenOrientation is unsupported on iOS")
         }
       : NativeModules.ArtsyNativeModule.lockActivityScreenOrientation,
+  clearCache:
+    Platform.OS === "ios"
+      ? () => {
+          console.error("clearCache is not yet supported on iOS")
```
Comment (pvinis) [inline src/app/NativeModules/ArtsyNativeModule.tsx:48]:
```suggestion
          console.error("clearCache is not needed on iOS. See HACKS.md.")
```

**Review by pvinis — APPROVED** [review]

**Comment by lordkiz** [comment]
> wow. blows my mind that this workaround works. i guess that crop lib is wonky. nice work kiz!
> 
> maybe change the PR desc to `This PR is about <jira link>`, so that it doesnt close it, and we can use this to track this issue? 🤔

Thanks, it needs to be tested on many devices. Does it work fine in yours?

**Comment by pvinis** [comment]
mine was working fine even before this, but ill give it a go now with this PR.

**Review by brainbicycle — DISMISSED** [review]
This looks okay to me, thanks for putting in the change! Unfortunately it is still happening for me after this change, maybe we can look at using a different library for the long term fix but might be worth merging this anyway if it solves it for some users.

**Review by MounirDhahri — APPROVED** [review]

**Comment by MounirDhahri** [comment]
Should we merge this already after fixing the merge conflicts?

**Comment by brainbicycle** [comment]
I will go ahead and merge but unfortunately I do think this is still a bug on some devices

**Pull request merged**